From cfd014bc5acfb5e5663e132d334ecb32634ad080 Mon Sep 17 00:00:00 2001
From: Gaku Inami <gaku.inami.xw@bp.renesas.com>
Date: Wed, 8 Feb 2017 11:28:22 +0900
Subject: [PATCH 1712/2066] Revert "mmc: tmio: Remove redundant clock handling
 when suspend"

This reverts commit 7ecbd9bb357b ("mmc: tmio: Remove redundant clock
handling when suspend")

For the suspend/resume function is revised in sdhi/mmc driver,
the conventional implementation will be reverted once.

Signed-off-by: Gaku Inami <gaku.inami.xw@bp.renesas.com>
[zou:Original patch taken from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git
 v4.9/rcar-3.5.1]
Signed-off-by: zou cao <cao.zou@windriver.com>
---
 drivers/mmc/host/tmio_mmc_pio.c |   10 ++++++++--
 1 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/drivers/mmc/host/tmio_mmc_pio.c b/drivers/mmc/host/tmio_mmc_pio.c
index 460b842..c18bdc6 100644
--- a/drivers/mmc/host/tmio_mmc_pio.c
+++ b/drivers/mmc/host/tmio_mmc_pio.c
@@ -1446,8 +1446,10 @@ EXPORT_SYMBOL(tmio_mmc_host_runtime_resume);
 int tmio_mmc_host_suspend(struct device *dev)
 {
 	struct mmc_host *mmc = dev_get_drvdata(dev);
+	struct tmio_mmc_host *host = mmc_priv(mmc);
 
-	tmio_mmc_hw_reset(mmc);
+	if (host->clk_disable)
+		host->clk_disable(host);
 
 	return 0;
 }
@@ -1455,7 +1457,11 @@ EXPORT_SYMBOL(tmio_mmc_host_suspend);
 
 int tmio_mmc_host_resume(struct device *dev)
 {
-	/* Empty for now */
+	struct mmc_host *mmc = dev_get_drvdata(dev);
+	struct tmio_mmc_host *host = mmc_priv(mmc);
+
+	tmio_mmc_clk_enable(host);
+
 	return 0;
 }
 EXPORT_SYMBOL(tmio_mmc_host_resume);
-- 
1.7.5.4

